context.go

```diff
@@ -102,36 +102,47 @@ func (context *Context) run(obj *wafObject, timeout time.Duration, cgoRefs *cgoR
 	ret := wafLib.wafRun(context.cContext, obj, result, uint64(timeout/time.Microsecond))
 
 	context.totalRuntimeNs.Add(result.total_runtime)
-	matches, actions, err := unwrapWafResult(ret, result)
+	events, actions, derivatives, err := unwrapWafResult(ret, result)
 	if err == ErrTimeout {
 		context.timeoutCount.Inc()
 	}
 
-	return matches, actions, err
+	return Result{
+		Events:      events,
+		Actions:     actions,
+		Derivatives: derivatives,
+	}, err
 }
 
-func unwrapWafResult(ret wafReturnCode, result *wafResult) (matches []byte, actions []string, err error) {
+func unwrapWafResult(ret wafReturnCode, result *wafResult) (matches []interface{}, actions []string, derivatives map[string]interface{}, err error) {
```

Comment:
Can you choose once and for all between "matches" and "events"? The `Result` time is about events but here we talk about matches

Reply:
I would definitively make this function return `Result` also

Reply:
aa96a06e42dcf737a57ce99253b77b3be1852fd3